### PR TITLE
feat: allow aux funds for name registration

### PR DESCRIPTION
**Pull request by gareys-cb**
**What changed? Why?**

This PR enables wallets using [EIP-5792](https://eips.ethereum.org/EIPS/eip-5792) with the `auxiliaryFunds` capability to register Base names even when the wallet is unfunded.

Currently, the "Register name" button is disabled.
Screenshot 2024-10-12 at 12 01 04 AM

With this change, wallets with the `auxiliaryFunds` capability won't be restricted from buying when they have an insufficient wallet balance.
Screenshot 2024-10-12 at 12 01 57 AM

https://github.com/user-attachments/assets/863250a7-b1e2-4038-bcaf-b65f576a637b

**Notes to reviewers**

Try it out locally or watch the videos. I've run the tests listed below locally.

**How has it been tested?**

Test with a wallet without capabilities to see that the button is disabled when wallet is unfunded.
Test with CB Smart Wallet which has capabilities to see that the button is enabled when wallet is unfunded.
Test with any funded wallet to see that it works when wallet is funded.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/2</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/components/Basenames/RegistrationForm/index.tsx [hunk 1]
```diff
@@ -395,7 +400,6 @@ export default function RegistrationForm() {
             baseSingleYearEthCost={singleYearBasePrice}
             isOpen={premiumExplainerModalOpen}
             toggleModal={togglePremiumExplainerModal}
-            nameLength={selectedName?.length}
```
Comment (brendan-defi) [inline apps/web/src/components/Basenames/RegistrationForm/index.tsx:398]: why are we removing this?

File: apps/web/src/hooks/useIsAuxiliaryFundsEnabled.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { useMemo } from 'react';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+
+export function useIsAuxiliaryFundsEnabled(): boolean {
+  const { chainId } = useAccount();
+  const { data } = useCapabilities();
```
Comment (JFrankfurt) [inline apps/web/src/hooks/useIsAuxiliaryFundsEnabled.ts:7]: Should prob use @kirkas 's `apps/web/src/hooks/useCapabilitiesSafe.ts` here

```suggestion
  const { data } = useCapabilitiesSafe();
```

Comment (JFrankfurt) [reply apps/web/src/hooks/useIsAuxiliaryFundsEnabled.ts:7]: Looks like you'd need to add `auxiliaryFunds` to its check

Comment (kirkas) [reply apps/web/src/hooks/useIsAuxiliaryFundsEnabled.ts:7]: yes and feel free to modify useCapabilitiesSafe to give you a auxiliaryFunds boolean

File: apps/web/src/components/Basenames/RegistrationForm/index.tsx (hunk 1 above)
Comment (gareys-cb) [reply apps/web/src/components/Basenames/RegistrationForm/index.tsx:398]: The prop doesn't exist on the component. It's a typeError.

Comment (brendan-defi) [reply apps/web/src/components/Basenames/RegistrationForm/index.tsx:398]: makes sense, I thought we were using that but I guess it's now obsolete.

File: apps/web/src/hooks/useCapabilitiesSafe.ts [hunk 3]
```diff
@@ -12,26 +12,38 @@ import { Chain } from 'viem';
 import { useAccount } from 'wagmi';
 import { useCapabilities } from 'wagmi/experimental';
 
+// To add a new capability, add it to this list
+const capabilitiesList = ['atomicBatch', 'paymasterService', 'auxiliaryFunds'] as const;
```
Comment (JFrankfurt) [inline apps/web/src/hooks/useCapabilitiesSafe.ts:16]: nit:
```suggestion
const CAPABILITIES_LIST = ['atomicBatch', 'paymasterService', 'auxiliaryFunds'] as const;
```

**Review by JFrankfurt — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for JFrankfurt @ 2024-10-17 19:53:55 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by JFrankfurt — APPROVED** [review]

**Pull request closed without merging**